apps/extension/src/routes/popup/settings/settings-rpc.tsx

```diff
@@ -48,38 +68,34 @@ export const SettingsRPC = () => {
             <div className='flex flex-col items-center justify-center gap-2'>
               <div className='flex items-center gap-2 self-start'>
                 <div className='text-base font-bold'>RPC URL</div>
-                {rpcError ? (
-                  <div className={cn('italic', 'text-red-400')}>
-                    Failed to get response from rpc
-                  </div>
-                ) : null}
+                {rpcError ? <div className='italic text-red-400'>{rpcError}</div> : null}
               </div>
               <Input
                 variant={rpcError ? 'error' : 'default'}
-                value={rpc}
-                onChange={e => {
-                  setRpc(e.target.value);
-                  setRpcError(false);
+                value={rpcInput}
+                onChange={evt => {
+                  setRpcError(undefined);
+                  setRpcInput(evt.target.value);
                 }}
                 className='text-muted-foreground'
               />
             </div>
             <div className='flex flex-col gap-2'>
               <p className='font-headline text-base font-semibold'>Chain id</p>
               <div className='flex h-11 w-full items-center rounded-lg border bg-background px-3 py-2 text-muted-foreground'>
-                {chainId}
+                {newChainId ?? currentChainId}
               </div>
             </div>
           </div>
-          <Button
-            variant='gradient'
-            size='lg'
-            className='w-full'
-            type='submit'
-            disabled={rpc === grpcEndpoint || rpcError}
-          >
-            Save
-          </Button>
+          {countdownTime != null ? (
```

Comment:
```suggestion
          {countdownTime ? (
```

Reply:
the countdown currently displays a countdownTime of 0 at the end. if i made this change it would at that moment revert to a normal button again